mev_inspect/aave_liquidations.py

```diff
@@ -0,0 +1,113 @@
+from typing import List
+
+from mev_inspect.schemas.classified_traces import (
+    Classification,
+    ClassifiedTrace,
+    # CallTrace,
+    DecodedCallTrace,
+)
+from mev_inspect.schemas.liquidations import Liquidation
+
+
+# Inspect list of classified traces and identify liquidation
+def get_liquidations(traces: List[ClassifiedTrace]):
+    tx = []
+    liquidations = []
+    result = []
+
+    # Protocol contract address must be in included, below is AaveLendingPoolCoreV1
+    addrs = [
+        "0x3dfd23A6c5E8BbcFc9581d2E864a68feb6a076d3",
+        "0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9",
+        "0x398eC7346DcD622eDc5ae82352F02bE94C62d119",
+    ]
+
+    # Used to remove double-counted 'from' transfers
+    from_doubles = []
+    transfers_to = []
+    transfers_from = []
+
+    # For each trace
+    for trace in traces:
+
+        if isinstance(trace, DecodedCallTrace):
+
+            # Check for liquidation and register trace and unique liquidator
+            if (
+                trace.classification == Classification.liquidate
+                and trace.transaction_hash not in tx
+            ):
+
+                liquidations.append(trace)
+                addrs.append(trace.from_address)
+                tx.append(trace.transaction_hash)
+
+                # Found liquidation, now parse inputs for data
+                for input in trace.inputs:
+
+                    if input == "_purchaseAmount":
+                        liquidation_amount = trace.inputs[input]
+                    elif input == "_collateral":
+                        collateral_address = trace.inputs[input]
+                        # This will be the address of the sent token
+                    elif input == "_reserve":
+                        reserve = trace.inputs[input]
+                        # This will be the address of the received token
+                    elif input == "_user":
+                        liquidated_usr = trace.inputs[input]
+                # Register liquidation
+                result.append(
```

Comment:
based on how the loop is written, if you never get a “_user” input, you’ll get a variable not defined error for liquidated usr here (same for the other variables)

Reply:
How can there not be a _user input?

Reply:
if you never see that key

like logically in the code there is a case where you never hit that if branch

Reply:
I'm not sure how to address this

Reply:
`liquidated_user` the variable in python won't exist of there is no `_user` value

so when it's used on 57 you'll get a variable doesn't exist exception